src/net.cpp

```diff
@@ -2957,7 +2957,7 @@ CNode::CNode(NodeId idIn, ServiceFlags nLocalServicesIn, int nMyStartingHeightIn
     m_conn_type(conn_type_in),
     nLocalServices(nLocalServicesIn),
     nMyStartingHeight(nMyStartingHeightIn),
-    m_inbound_onion(inbound_onion)
+    m_inbound_onion(inbound_onion && conn_type_in == ConnectionType::INBOUND)
```

Comment:
I might be misunderstanding but in what case would the caller pass `inbound_onion` parameter as true but lie about this, as it is an outbound connection?
Wouldn't this be better as an assertion?

Reply:
Or an `Assume`, if `Assert` is too strong and risky

Reply:
Good idea, `inbound_onion` should only be true when a new CNode is instantiated from `CConnman::AcceptConnection()`, so this should only fail if a bug is added. Made it an assert.

Reply:
> in what case would the caller pass `inbound_onion` parameter as true but lie about this

So, adding the assertion found one! Our net_tests had a case that lied; updating it.

Reply:
Hm, a fuzzer needs to be updated, too.